ios/Sources/AutoLayoutView.swift

```diff
@@ -169,15 +177,20 @@ import UIKit
                     nextCell.frame.origin.y = maxBound
                 }
                 if isNextCellVisible {
-                    maxBoundNextCell = max(maxBound, nextCellBottom)
+                    maxBoundNextCell = max(maxBound, nextCell.frame.maxY)
                 }
             }
+            updateLastMaxBoundOverall(currentCell: cellContainer, nextCell: nextCell)
         }
 
         lastMaxBound = maxBoundNextCell
         lastMinBound = minBound
     }
 
+    private func updateLastMaxBoundOverall(currentCell: CellContainer, nextCell: CellContainer) {
+        lastMaxBoundOverall = max(lastMaxBoundOverall, horizontal ? currentCell.frame.maxX : currentCell.frame.maxY, horizontal ? nextCell.frame.maxX : nextCell.frame.maxY)
```

Comment:
nit: this is a bit hard to read, let's split it in multiple lines:
```suggestion
        lastMaxBoundOverall = max(
            lastMaxBoundOverall, 
            horizontal ? currentCell.frame.maxX : currentCell.frame.maxY, 
            horizontal ? nextCell.frame.maxX : nextCell.frame.maxY
       )
```